anomalib/data/utils/split.py

```diff
@@ -11,76 +11,97 @@
+def concatenate_datasets(datasets: Sequence[AnomalibDataset]) -> AnomalibDataset:
+    """Concatenate multiple datasets into a single dataset object.
 
     Args:
-        samples (DataFrame): Dataframe containing dataset info such as filenames, splits etc.
-        seed (int, optional): Random seed to ensure reproducibility. Defaults to 0.
-        normal_label (str): Name of the normal label. For MVTec AD, for instance, this is normal_label.
-    """
-
-    if seed is not None:
-        random.seed(seed)
+        datasets (Sequence[AnomalibDataset]): Sequence of at least two datasets.
 
-    # Split normal images.
-    normal_test_image_indices = samples.index[(samples.split == "test") & (samples.label == normal_label)].to_list()
-    num_normal_valid_images = len(normal_test_image_indices) // 2
+    Returns:
+        AnomalibDataset: Dataset that contains the combined samples of all input datasets.
+    """
+    concat_dataset = datasets[0]
+    for dataset in datasets[1:]:
+        concat_dataset += dataset
+    return concat_dataset
 
-    indices_to_sample = random.sample(population=normal_test_image_indices, k=num_normal_valid_images)
-    samples.loc[indices_to_sample, "split"] = "val"
 
-    # Split abnormal images.
-    abnormal_test_image_indices = samples.index[(samples.split == "test") & (samples.label != normal_label)].to_list()
-    num_abnormal_valid_images = len(abnormal_test_image_indices) // 2
+def random_split(
+    dataset: AnomalibDataset, split_ratio: Union[float, Sequence[float]], label_aware: bool = False
+) -> List[AnomalibDataset]:
+    """Perform a random split of a dataset.
 
-    indices_to_sample = random.sample(population=abnormal_test_image_indices, k=num_abnormal_valid_images)
-    samples.loc[indices_to_sample, "split"] = "val"
+    Args:
+        dataset (AnomalibDataset): Source dataset
+        split_ratio (Union[float, Sequence[float]]): Fractions of the splits that will be produced. The values in the
+            sequence must sum to 1. If a single value is passed, the ratio will be converted to
+            [1-split_ratio, split_ratio].
+        label_aware (bool): When True, the relative occurrence of the different class labels of the source dataset will
+            be maintained in each of the subsets.
+    """
 
-    return samples
+    if isinstance(split_ratio, float):
+        split_ratio = [1 - split_ratio, split_ratio]
+
+    assert math.isclose(sum(split_ratio), 1) and sum(split_ratio) <= 1, "split ratios must sum to 1."
+    assert all(0 < ratio < 1 for ratio in split_ratio), "all split ratios must be between 0 and 1."
+
+    # create list of source data
+    if label_aware:
+        indices_per_label = [group.index for _, group in dataset.samples.groupby("label_index")]
+        per_label_datasets = [dataset.subsample(indices) for indices in indices_per_label]
+    else:
+        per_label_datasets = [dataset]
+
+    # split each (label-aware) subset of source data
+    subsets = []
```

Comment:
just to make it easier to follow
```suggestion
    # outter list: per-label unique, inner list: random subsets with the given ratio 
    subsets: List[List[AnomalibDataset]] = []
```